tracing/ops_tracing/_api.py

```diff
@@ -0,0 +1,213 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The tracing API for the charms."""
+
+from __future__ import annotations
+
+import logging
+
+import ops
+
+from ._buffer import Destination
+from .vendor.charms.certificate_transfer_interface.v1.certificate_transfer import (
+    CertificateTransferRequires,
+)
+from .vendor.charms.tempo_coordinator_k8s.v0.tracing import (
+    AmbiguousRelationUsageError,
+    ProtocolNotRequestedError,
+    TracingEndpointRequirer,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Tracing(ops.Object):
+    """Initialise the tracing service.
+
+    Usage:
+        - Include ``ops[tracing]`` in your dependencies.
+        - Declare the relations that the charm supports.
+        - Initialise ``Tracing`` with the names of these relations.
+
+    Example::
+
+        # charmcraft.yaml
+        requires:
+            charm-tracing:
+                interface: tracing
+                limit: 1
+                optional: true
+            send-ca-cert:
+                interface: certificate_transfer
+                limit: 1
+                optional: true
+
+        # src/charm.py
+        import ops.tracing
+
+        class SomeCharm(ops.CharmBase):
+            def __init__(self, framework: ops.Framework):
+                ...
+                self.tracing = ops.tracing.Tracing(
+                    self,
+                    tracing_relation_name="charm-tracing",
+                    ca_relation_name="send-ca-cert",
+                )
+
+    Args:
+        charm: your charm instance
+        tracing_relation_name: the name of the relation that provides the
+            destination to send trace data to.
+        ca_relation_name: the name of the relation that provides the CA
+            list to validate the tracing destination against.
+        ca_data: a fixed CA list (PEM bundle, a multi-line string).
+
+    If the destination is resolved to an HTTPS URL, a CA list is required
+    to establish a secure connection.
+
+    The CA list can be provided over a relation via the ``ca_relation_name``
+    argument, as a fixed string via the ``ca_data`` argument, or the system CA
+    list will be used if the earlier two are both ``None``.
+    """
+
+    def __init__(
+        self,
+        charm: ops.CharmBase,
+        tracing_relation_name: str,
+        *,
+        ca_relation_name: str | None = None,
+        ca_data: str | None = None,
+    ):
+        """Initialise the tracing service."""
+        super().__init__(charm, f'{tracing_relation_name}+{ca_relation_name}')
```

Comment:
this might be `"somethingNone"`, is that pretty? I'd rather it be a fixed `ops-tracing` (or `ops-tracing-tracing-obj` if you're afraid users might reuse that name) rather than something dynamic, easier to inspect when there's issues.

Reply:
Yes it can be `charm-tracing+None`, I'm OK with that.